ARCHITECTURE.md

```diff
@@ -0,0 +1,133 @@
+# Edge endpoint architecture
+
+## Overview
+
+The edge endpoint provides a way for clients to use the existing Groundlight API in a hybrid mode: most inference requests will be handled locally while all other requests will be forwarded to the Groundlight cloud service. This enables client programs to get low-latency results for most requests while still being able to use the full Groundlight API.
+
+## Structure
+
+The edge endpoint is implemented as a set of Kubernetes resources (defined by the helm chart in the [helm directory](deploy/helm/groundlight-edge-endpoint/)).
+
+There is a single pod for the main logic of the edge endpoint and one pod for each inference model.
```

Comment:
nit: by default there is one pod for each inference model, but if desired the number of inference server pod replicas can be scaled separately from the edge-endpoint pod.

Reply:
From my read of the code, there's no way for a user to ask for more than one replica of the inference pods.